### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by svartalf** [svartalf/rust-battop#5]
This is a tracking issue for Windows support.

Main problem is that [termion](https://crates.io/crates/termion) backend for [tui](https://crates.io/crates/tui) dependency does not support Windows terminals.

Another option for tui - [crossterm](https://crates.io/crates/crossterm) supports Windows and should be considered as a replacement.

**Comment by svartalf** [comment]
Possible switch to `crossterm` is blocked by this issue: https://github.com/TimonPost/crossterm/issues/81 (I think), since it is important for `battop` to handle `←`, `→` and `^C` key combinations.

**Comment by svartalf** [comment]
Note to myself: blocking PR was closed, nothing should block windows support now.

**Comment by svartalf** [comment]
And it is blocked by this PR now: https://github.com/fdehau/tui-rs/pull/150

**Comment by svartalf** [comment]
Almost done, blocked only because there is no `tui` crate release with the changes from this PR: https://github.com/fdehau/tui-rs/pull/153

**Comment by bew** [comment]
Hello, just discovered this nice tool! 
Is it still blocked? 🙄

**Comment by svartalf** [comment]
No, it is not blocked anymore, switch to `crossterm` backend for `tui` should be done, but I don't really have time to do that now.

PRs are more that welcome!

**Comment by ratijas** [comment]
Interesting to the point at which I'm almost willing to give it a try.

What was the whole point, again? Terminal UI library, which is being used now, provides too much capabilities that cannot be replicated on Windows, so devs just gave on supporting it, and thus it fails to even compile? So we need to move on to a less capable library which would support the lowest common denominator of all platforms. Am I getting this right?

**Comment by svartalf** [comment]
@ratijas the idea is to pick up terminal UI library, which will provide the same interface (or, at least the almost same) as in the current implementation, while also providing support for at least Linux, macOS and Windows. If it will also support *BSD family, that will be a great bonus.

I'm not aware about current state of `termion` crate, but the idea to switch to `crossterm` still stands.

Does this answers your question?
